pkg/wallet/wallet.go

```diff
@@ -409,6 +404,33 @@ func (c *Wallet) resolveCredentials(verify bool, credentials ...CredentialToPres
 	return response, nil
 }
 
+func (c *Wallet) verifyCredential(credential json.RawMessage) (bool, error) {
+	// TODO resolve stored DIDs which can not be resolved by context?
+	opts := verifiable.WithPublicKeyFetcher(
+		verifiable.NewDIDKeyResolver(c.ctx.VDRegistry()).PublicKeyFetcher(),
+	)
+
+	_, err := verifiable.ParseCredential(credential, opts)
+	if err != nil {
+		return false, fmt.Errorf("credential verification failed: %w", err)
+	}
+
+	return true, nil
+}
+
+// TODO should we check proof of each credentials inside presentation? under discussion in universal wallet 2020.
```

Comment:
Why wouldn't the proofs be checked?

Reply:
Yes, the point is should we check proof of each VCs inside presentation or we just have to check the holder signature in presentation. I am raising a question in universal wallet community since it is not clearly explained in spec. I will fix this in next PR once I get the response.

Reply:
We need to check both the VCs and the holder signature.

Reply:
fixed